runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/AbstractAwsChunkedByteReadChannel.kt

```diff
@@ -203,4 +204,48 @@ public abstract class AbstractAwsChunkedByteReadChannel(
         chunkOffset = 0
         return trailerBody
     }
+
+    /**
+     * Given a signing config, make a deep copy, changing the signatureType and hashSpecification configuration values
+     * to specify chunk signing.
+     * @param config the original signing config
+     * @return an [AwsSigningConfig] which can be used by a signer to sign chunks
+     */
+    private fun getChunkSigningConfig(config: AwsSigningConfig): AwsSigningConfig = AwsSigningConfig {
```

Comment:
**Style**: Consider an extension method (e.g., `AwsSigningConfig.toChunkSigningConfig`). Applies to `getTrailingHeadersSigningConfig` as well.